oshmem/mca/sshmem/ucx/sshmem_ucx_module.c

```diff
@@ -189,8 +189,18 @@ segment_create(map_segment_t *ds_buf,
             ds_buf->allocator   = &sshmem_ucx_allocator;
             return OSHMEM_SUCCESS;
         }
+        else if(hint) {
```

Comment:
If includes the possibility of  `hint == SHMEM_HINT_DEVICE_NIC_MEM`. What's the reasoning?